3/apollo.js

```diff
@@ -20,85 +18,156 @@ const typeDefs = gql`
 
   type Assignee {
     id: ID!
-    name: String
+    name: String!
+	password: String!
   }
 
   type Query {
 	todo(id: ID!): Todo
-    todos(assignee: ID): [Todo]
+    todos(assignee: ID, first: Int, offset: Int, desc: Boolean): [Todo]
   }
   
   type Mutation{
 	  login(userName: String!, pwd: String!): String! 
 	  createTodo(message: String, assignee: ID!): Todo
 	  updateTodo(id: ID!, message: String): Todo
 	  deleteTodo(id: ID!): Todo
-	  createAssignee(name: String!): Assignee
+	  createAssignee(name: String!, password: String!): Assignee
+	  updateAssignee(id: ID!, name: String!, password: String!): Assignee
   }
 `;
 
-const getAssignees= ()=> [
-      {id: '1', name: 'Hans'},
-      {id: '2', name: 'Hanna'}
-    ];
-
-const getTodos = ()=>[
-      { id: '1', message: 'Foo', assignee: assignees[0],},
-      { id: '2', message: 'Bar', assignee: assignees[1],},
-      { id: '3', message: 'Baz', assignee: assignees[0],}
-    ];
-
 function getRandomId(){
 	return Math.floor(Math.random() * Math.floor(9999999)).toString();
 }	
 	
 const resolvers = {
   Query: {
 	todo: (parent, args, context, info) => {
-		return todos.find((todo)=>{return todo.id == args.id});
+		return neo4jgraphql(parent, args, context, info)
 	},
-    todos: (parent, args, context, info) => {
-		let result = todos;
+    todos: async(parent, args, context, info) => {
+		const session = context.driver.session()
+		let result
 		if(args.assignee != null){
-			result = todos.filter((todo)=>{return todo.assignee.id == args.assignee});
+			let query = `
+			MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person {id: $assignee})
+			RETURN t, p
+			ORDER BY p.name
+			SKIP $s
+			LIMIT $l
+			`
+			if(args.desc == true){
+				query = `
+				MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person {id: $assignee})
+				RETURN t, p
+				ORDER BY p.name DESC
+				SKIP $s
+				LIMIT $l
+				`
+			}
+			result = await session.run(query,
+			{
+				assignee: args.assignee,
+				s: args.offset,
+				l: args.first
+			})
+		}else{
+			let query = `
+			MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person)
+			RETURN t, p
+			ORDER BY p.name
+			SKIP $s
+			LIMIT $l
+			`
+			if(args.desc == true){
+				query = `
+				MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person)
+				RETURN t, p
```

Comment:
```suggestion
				RETURN t { .*, asssignee: properties(p) }
```
Then you can do
```
			let response = record.get('t')
```
on line 191 :point_down: